File: src/main/java/com/example/chatService/dto/ParticipantDto.java
package com.example.chatService.dto;

import com.example.chatService.entity.Participant;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class ParticipantDto {
    private String userId;
    private String username;
    private RoomRole role;

    public static ParticipantDto from(Participant participant) {
        return new ParticipantDto(
                participant.getUserId(),
                null,
                participant.getRole()
        );
    }
}

File: src/main/java/com/example/chatService/dto/ParticipantDto.kt
package com.example.chatService.dto

import com.example.chatService.entity.Participant

data class ParticipantDto(
    val userId: String,
    val username: String?,
    val role: RoomRole
) {
    companion object {
        fun from(participant: Participant): ParticipantDto {
            return ParticipantDto(
                userId = participant.userId,
                username = null,
                role = participant.role!!
            )
        }
    }
}
